sdks/java/core/src/test/java/org/apache/beam/sdk/testing/FileChecksumMatcherTest.java

```diff
@@ -89,16 +99,96 @@ public void testMatcherVerifySingleFile() throws IOException{
   }
 
   @Test
-  public void testMatcherVerifyMultipleFilesInOneDir() throws IOException {
-    File tmpFile1 = tmpFolder.newFile();
-    File tmpFile2 = tmpFolder.newFile();
+  public void testMatcherThatVerifiesMultipleFiles() throws IOException {
+    File tmpFile1 = tmpFolder.newFile("result-000-of-002");
+    File tmpFile2 = tmpFolder.newFile("result-001-of-002");
+    File tmpFile3 = tmpFolder.newFile("tmp");
     Files.write("To be or not to be, ", tmpFile1, StandardCharsets.UTF_8);
     Files.write("it is not a question.", tmpFile2, StandardCharsets.UTF_8);
+    Files.write("tmp", tmpFile3, StandardCharsets.UTF_8);
+
     FileChecksumMatcher matcher =
         new FileChecksumMatcher(
             "90552392c28396935fe4f123bd0b5c2d0f6260c8",
+            IOChannelUtils.resolve(tmpFolder.getRoot().getPath(), "result-*"));
+
+    assertThat(pResult, matcher);
+  }
+
+  @Test
+  public void testMatcherThatVerifiesFileWithEmptyContent() throws IOException {
+    File emptyFile = tmpFolder.newFile("result-000-of-001");
+    Files.write("", emptyFile, StandardCharsets.UTF_8);
+    FileChecksumMatcher matcher =
+        new FileChecksumMatcher(
+            "da39a3ee5e6b4b0d3255bfef95601890afd80709",
             IOChannelUtils.resolve(tmpFolder.getRoot().getPath(), "*"));
 
     assertThat(pResult, matcher);
   }
+
+  @Test
+  public void testReadWithRetriesFailsWhenTemplateIncorrect()
```

Comment:
You have a test here that verifies incorrect template behavior, but no test which verifies any template other than the default template. Probably a good idea to add one?